provider/maas/environ.go

```diff
@@ -2120,8 +2120,15 @@ func (env *maasEnviron) allocateContainerAddresses1(hostInstanceID instance.Id,
 	for i, nic := range preparedInfo {
 		maasNICID := ""
 		nameToParentName[nic.InterfaceName] = nic.ParentInterfaceName
+		nicVLANID, knownSubnet := subnetCIDRToVLANID[nic.CIDR]
 		if nic.InterfaceName != primaryNICName {
-			nicVLANID := subnetCIDRToVLANID[nic.CIDR]
+			if !knownSubnet {
+				logger.Warningf("NIC %v has no subnet - setting to manual and using untagged VLAN", nic.InterfaceName)
```

Comment:
Surprisingly more often than we originally though it might - lots of users, including our current CI setup, use the second NIC only on dual-NIC nodes, and leave the first NIC unconfigured.

However, thinking about it now, I think we should perhaps log this at Info level, and add a check to ensure *at least one* NIC of the device gets an address from MAAS, otherwise networking will be broken inside the container later - if it even manages to provision in the first place.